generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/annotations/GraphQLAbstractType.kt

```diff
@@ -0,0 +1,6 @@
+package com.expediagroup.graphql.generator.annotations
+
+import kotlin.reflect.KClass
+
+@Target(AnnotationTarget.PROPERTY, AnnotationTarget.FUNCTION)
+annotation class GraphQLAbstractType(val possibleTypes: Array<KClass<*>>, val name: String, val description: String)
```

Comment:
guess technically this would apply also to interfaces but since they have to declare some common fields -> should this be just applicable to the unions?

Reply:
Yea, as I was walking through more required changes, it is going to only work with unions. We will need a different annotation that we can put on both the classes and the return type to implement custom interfaces